### ISSUE TITLE
EXOTransportRule Enabled parameter doesn't work

### ISSUE THREAD

**Issue by jeremyhagan** [microsoft/Microsoft365DSC#3464]
#### Details of the scenario you tried and the problem that is occurring
When describing a Transport rule in a Configuration the setting for Enabled makes no difference. When returning the current configuration, the value of Enabled is always $null, regardless of the configuration of the target rule. Setting $True, $False, or $null in the Configuration does nothing.

#### Verbose logs showing the problem
This is the log when the target resource is ENABLED
`VERBOSE: [DSCRunner]:                            [[EXOTransportRule]EXOTransportRule-Reject: Reject outbound overclassified messages, HEADER] Current Values: ActivationDate=$null
ADComparisonAttribute=$null
ADComparisonOperator=$null
AddManagerAsRecipientType=$null
AddToRecipients=$null
AnyOfCcHeader=$null
AnyOfCcHeaderMemberOf=$null
AnyOfRecipientAddressContainsWords=$null
AnyOfRecipientAddressMatchesPatterns=$null
AnyOfToCcHeader=$null
AnyOfToCcHeaderMemberOf=$null
AnyOfToHeader=$null
AnyOfToHeaderMemberOf=$null
ApplicationId=***
ApplyClassification=$null
ApplyHtmlDisclaimerFallbackAction=$null
ApplyHtmlDisclaimerLocation=$null
ApplyHtmlDisclaimerText=$null
ApplyOME=False
ApplyRightsProtectionCustomizationTemplate=$null
ApplyRightsProtectionTemplate=$null
AttachmentContainsWords=$null
AttachmentExtensionMatchesWords=$null
AttachmentHasExecutableContent=False
AttachmentIsPasswordProtected=False
AttachmentIsUnsupported=False
AttachmentMatchesPatterns=$null
AttachmentNameMatchesPatterns=$null
AttachmentProcessingLimitExceeded=False
AttachmentPropertyContainsWords=$null
AttachmentSizeOver=$null
BetweenMemberOf1=$null
BetweenMemberOf2=$null
BlindCopyTo=$null
CertificatePassword=$null
CertificatePath=***
CertificateThumbprint=***
Comments=$null
ContentCharacterSetContainsWords=$null
CopyTo=$null
Credential=$null
DeleteMessage=False
DlpPolicy=$null
Enabled=$null
Ensure=Present`

This is the log when the target resource is DISABLED
`VERBOSE: [DSCRunner]:                            [[EXOTransportRule]EXOTransportRule-Reject: Reject outbound overclassified messages, HEADER] Current Values: ActivationDate=$null
ADComparisonAttribute=$null
ADComparisonOperator=$null
AddManagerAsRecipientType=$null
AddToRecipients=$null
AnyOfCcHeader=$null
AnyOfCcHeaderMemberOf=$null
AnyOfRecipientAddressContainsWords=$null
AnyOfRecipientAddressMatchesPatterns=$null
AnyOfToCcHeader=$null
AnyOfToCcHeaderMemberOf=$null
AnyOfToHeader=$null
AnyOfToHeaderMemberOf=$null
ApplicationId=***
ApplyClassification=$null
ApplyHtmlDisclaimerFallbackAction=$null
ApplyHtmlDisclaimerLocation=$null
ApplyHtmlDisclaimerText=$null
ApplyOME=False
ApplyRightsProtectionCustomizationTemplate=$null
ApplyRightsProtectionTemplate=$null
AttachmentContainsWords=$null
AttachmentExtensionMatchesWords=$null
AttachmentHasExecutableContent=False
AttachmentIsPasswordProtected=False
AttachmentIsUnsupported=False
AttachmentMatchesPatterns=$null
AttachmentNameMatchesPatterns=$null
AttachmentProcessingLimitExceeded=False
AttachmentPropertyContainsWords=$null
AttachmentSizeOver=$null
BetweenMemberOf1=$null
BetweenMemberOf2=$null
BlindCopyTo=$null
CertificatePassword=$null
CertificatePath=***
CertificateThumbprint=***
Comments=$null
ContentCharacterSetContainsWords=$null
CopyTo=$null
Credential=$null
DeleteMessage=False
DlpPolicy=$null
Enabled=$null
Ensure=Present`

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
Configuration EXOConfig
{
    param (
    )

    $OrganizationName = $ConfigurationData.NonNodeData.OrganizationName

    Import-DscResource -ModuleName 'Microsoft365DSC' -ModuleVersion '1.23.628.1'

    Node localhost
    {
        EXOTransportRule "EXOTransportRule-Reject: Reject outbound overclassified messages, HEADER"
        {
            ApplicationId                             = $ConfigurationData.NonNodeData.ApplicationId;
            ApplyOME                                  = $False;
            AttachmentHasExecutableContent            = $False;
            AttachmentIsPasswordProtected             = $False;
            AttachmentIsUnsupported                   = $False;
            AttachmentProcessingLimitExceeded         = $False;
            CertificateThumbprint                     = $ConfigurationData.NonNodeData.CertificateThumbprint;
            DeleteMessage                             = $False;
			Enabled									  = $True;
            Ensure                                    = "Present";
            ExceptIfAttachmentHasExecutableContent    = $False;
            ExceptIfAttachmentIsPasswordProtected     = $False;
            ExceptIfAttachmentIsUnsupported           = $False;
            ExceptIfAttachmentProcessingLimitExceeded = $False;
            ExceptIfHasNoClassification               = $False;
            ExceptIfHasSenderOverride                 = $False;
            FromScope                                 = "InOrganization";
            HasNoClassification                       = $False;
            HasSenderOverride                         = $False;
            HeaderMatchesMessageHeader                = "x-foo";
            HeaderMatchesPatterns                     = @("FOO");
            Mode                                      = "Audit";
            ModerateMessageByManager                  = $False;
            Name                                      = "Reject: Reject outbound overclassified messages, HEADER";
            Priority                                  = 10;
            Quarantine                                = $False;
            RecipientAddressType                      = "Resolved";
            RejectMessageEnhancedStatusCode           = "5.7.1";
            RejectMessageReasonText                   = "Message rejected due to inappropriate protective markings. Please only use OFFICIAL:Sensitive or below.";
            RemoveOME                                 = $False;
            RemoveOMEv2                               = $False;
            RemoveRMSAttachmentEncryption             = $False;
            RouteMessageOutboundRequireTls            = $False;
            RuleErrorAction                           = "Defer";
            RuleSubType                               = "None";
            SenderAddressLocation                     = "Header";
            StopRuleProcessing                        = $False;
            TenantId                                  = $OrganizationName;
        }
    }
}
```

#### The operating system the target node is running
OsName               : Microsoft Windows Server 2019 Datacenter
OsOperatingSystemSKU : DatacenterServerEdition
OsArchitecture       : 64-bit
WindowsVersion       : 1809
WindowsBuildLabEx    : 17763.1.amd64fre.rs5_release.180914-1434
OsLanguage           : en-US
OsMuiLanguages       : {en-US}

#### Version of the DSC module that was used ('dev' if using current dev branch)
1.23.628.1

**Comment by tom-sutherland** [comment]
I recently ran into the same issue myself - a few thoughts:

The problem seems to be that Get-TargetResource is looking for an **Enabled** property with a true/false value in response to Get-TransportRule. Get-TransportRule doesn't return an Enabled property, but does return **State** with a value of Enabled or Disabled.

To avoid a breaking change, I wonder if it's worth converting the $TransportRule.state string into a true/false boolean, as follows:

```
        if ($TransportRule.State -eq 'Enabled') {
            $result.Enabled = $true
        }
        elseif($TransportRule.State -eq 'Disabled') {
            $result.Enabled = $false
        }

```

There is a related problem whereby setting config on an existing transport rule does not enable/disable according to your config. I think this is because you can't set the enabled state via Set-TransportRule and instead need to use Enable-TransportRule/Disable-TransportRule.

So we'd need logic something like this in the 'different values' case:

```
        if ($Enabled -eq $false -and $currentTransportRuleConfig.Enabled -eq $true) {
            Disable-TransportRule -Identity $Name -Confirm:$false
             Write-Verbose "Transport rule '$($Name)' is enabled but should be disabled, disabling now."
        }
        elseif($Enabled -eq $true -and $currentTransportRuleConfig.Enabled -eq $false) {
            Enable-TransportRule -Identity $Name -Confirm:$false
            Write-Verbose "Transport rule '$($Name)' is disabled but should be enabled, enabling now."
        }  
```

**Comment by andikrueger** [comment]
@tom2263 thank you for the analysis. This sound plausible and avoiding a breaking change is a good idea. Otherwise a fix for this issue would need to wait until the first release in October.

**Comment by titlerequired** [comment]
Just to add, I was testing some transport rules. They were created in disabled state directly in 365. I exported the DSC Config, deleted the rules and then applied them via DSC, the result was they were created & enabled. I can't see 'State' or 'Enabled' as part of the configuration that is exported.

**Comment by jeremyhagan** [comment]
It would be nice if DSC was consistent with the way the portal creates rules as disabled by default.

**Comment by malauter** [comment]
This was fixed in release v1.23.1227.1
